### PR TITLE
PowerShell Cmdlets for Azure HPC Cache service - API version 2020-03-01

### PR DESCRIPTION
**Pull request by romahamu**
## Description
Cherry picked commits from https://github.com/Azure/azure-powershell-pr/commits/Az.StorageCache

aeb14acc6b00da424d1f39fa2ef8f788bc9e49d9  Add mapping rule for HPCCache
c2112d74f6fca0fbbd81a2a35c39beac6321c2a4 Release-2020-04-28-HPCCache
00076831fb216fa42cf8e01cbf2fc39dd2c47004 Change StorageCache folder name to HPCCache and modify Upgrade cache cmdlet
fe870353bbae7d13ea9bfdaae85f7bcf02798338 [HPC-Cache] On-boarding of PowerShell Cmdlets for HPC Cache service. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/477

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/HPCCache/HPCCache/Az.HPCCache.psd1 [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        ProjectUri = 'https://github.com/Azure/azure-powershell'
+
+        # A URL to an icon representing this module.
+        # IconUri = ''
+
+        # ReleaseNotes of this module
+        ReleaseNotes = '* Preview of ''Az.HPCCache'' module'
+
+        # Prerelease string of this module
+        Prerelease = 'preview'
```
Comment (isra-fel) [inline src/HPCCache/HPCCache/Az.HPCCache.psd1:117]:
```suggestion
        # Prerelease = 'preview'
```
Preview is unnecessary if the module is to be published to the public gallery and version < 1.0.0

File: src/HPCCache/HPCCache/help/Update-AzHpcCache.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+# Update-AzHpcCache
+
+## SYNOPSIS
+Updates a HPC Cache.
+
+## SYNTAX
+
+### ByFieldsParameterSet (Default)
+```
+Update-AzHpcCache -ResourceGroupName <String> -Name <String> [-Upgrade] [-Flush] [-Force] [-PassThru] [-AsJob]
```
Comment (isra-fel) [inline src/HPCCache/HPCCache/help/Update-AzHpcCache.md:17]: Can `-Upgrade` and `-Flush` be used together? If not, I recommend split them into different parameter sets.

File: src/HPCCache/HPCCache/Az.HPCCache.psd1 (hunk 1 above)
Comment (romahamu) [reply src/HPCCache/HPCCache/Az.HPCCache.psd1]: Fixed in b2d9a68

File: src/HPCCache/HPCCache/help/Update-AzHpcCache.md (hunk 2 above)
Comment (romahamu) [reply src/HPCCache/HPCCache/help/Update-AzHpcCache.md]: Fixed in 0bfa666

**Pull request merged**